### DISCUSSION TITLE
Upload of Devices.csv

### DISCUSSION THREAD

**Discussion by anjota80** [netalertx/NetAlertX#613] [Ideas]
The current version allows the download of the devices.csv file, however, I cannot find a way to transfer the devices.csv to the config folder due to access issues on the current build. I have tried terminal, desktop drag drop and FileZilla etc.
Ideally in the configuration, you could select the location of the directory to upload the devices.csv file. For example, I have copied to the home Documents folder, so would be good to select the upload from that directory.

**Comment by jokob-sk** [comment]
Hi there,

Could you please share more details about the access issues you have?

As a workaround, have you tried mounting the pialert.conf and devices.csv files separately in your docker-compose?

**Comment by anjota80** [comment]
Hi, I have attached some screenshots. Images A and B show the access issues when trying to drag drop the devices.csv. I get the same issues when using terminal commands. I have even tried 'sudo su' to override the access.
Image C shows the compose file I am using from https://pimylifeup.com/raspberry-pi-pialert/. I have also tried the examples in the following link https://github.com/jokob-sk/Pi.Alert/blob/main/dockerfiles/README.md

![A](https://github.com/jokob-sk/Pi.Alert/assets/53611664/3d2d815c-8e12-4658-88e5-33997ef5d0c7)
![B](https://github.com/jokob-sk/Pi.Alert/assets/53611664/09132a81-cd74-4a23-9701-2b5602053f74)
![C](https://github.com/jokob-sk/Pi.Alert/assets/53611664/df17c996-ba91-4aaf-bd18-01fd028e7eba)

**Reply by jokob-sk** [reply]
Thank you for the details.

Under volumes try:

`./data/config/pialert.conf:/home/pi/pialert/config/pialert.conf`

`./data/config/devices.csv:/home/pi/pialert/config/devices.csv`

And remove the config folder mapping. You can revert back after the import.

**Reply by jokob-sk** [reply]
Make sure you have the files in the right spot

**Reply by anjota80** [reply]
Hi, I have completed a fresh install using the above but still have the Permission Denied issue on the main folder. See the images with the new Compose file with your suggestions above. I also cannot access the pialert.conf file to update the subnets now.

As I am testing a lot with Pi.Alert and creating fresh installs, I just need a simple way to upload the devices.csv file to save me manually changing the device information after each install.

![A](https://github.com/jokob-sk/Pi.Alert/assets/53611664/96d95acb-8c83-4d89-82e9-107ab233f4a6)

![B](https://github.com/jokob-sk/Pi.Alert/assets/53611664/5eb1ac9c-bf69-412e-8534-6ecf841e558b)

**Reply by anjota80** [reply]
OK, so after about an hour playing around with access and permissions, I have finally managed to move the devices.csv to the config folder. I had to go to the terminal and enter the following:

**su – root
chmod 777 /opt/stacks/pialert
chmod 777 /opt/stacks/pialert/data
chmod 777 /opt/stacks/pialert/data/config**

I then used FileZilla to copy the devices.csv from my PC to the Documents folder on my Raspberry Pi. Then I used the Raspberry Pi desktop to drag and drop the devices.csv to /opt/stacks/pialert/data/config. I then selected the CSV Import option within Pi.Alert and it worked. As you can see this is very time consuming. Ideally within the Pi.Alert settings, you could select the upload location of the devices.csv either direct from the home folders or from a remote PC. Perhaps add this option to the next release? 

![C](https://github.com/jokob-sk/Pi.Alert/assets/53611664/ad1cc166-faa7-441b-a813-c60d90bfcbf2)

**Reply by jokob-sk** [reply]
Glad to hear you managed to get the permissions issue resolved 👍 I will consider your feature request in future if I get more people with similar requests.

**Reply by anjota80** [reply]
Thank you
